specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,267 @@
+      SchemaContentType,
+    NoContentResponse<SchemaHeaders>
+  >;
+}
+
+// Models
+
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+union SchemaContentTypeValues {
+  @added(ServiceApiVersions.V2022_10)
+  customBinary: "application/octet-stream",
+
+  @added(ServiceApiVersions.V2022_10)
+  custom: "text/plain; charset=utf-8",
+
+  avro: "application/json; serialization=Avro",
+
+  @added(ServiceApiVersions.V2022_10)
+  json: "application/json; serialization=Json",
+
+  @added(ServiceApiVersions.V2023_07_01)
+  protobuf: "text/vnd.ms.protobuf",
+}
+
+@doc("The content type for given schema. Each schema type has an associated content-type.")
+model SchemaContentType {
+  @doc("The content type for given schema.")
+  @header("Content-Type")
+  contentType: SchemaContentTypeValues;
+}
+
+@doc("Content of the schema.")
+model SchemaContent {
+  @body
+  @doc("String representation (UTF-8) of the schema.")
+  content: bytes;
+}
+
+@doc("Headers for metadata of a schema.")
+model SchemaHeaders {
+  @doc("URL location of schema, identified by schema group, schema name, and version.")
+  @header("Location")
+  location: string;
+
+  @doc("References specific schema in registry namespace.")
+  @header("Schema-Id")
+  schemaId: string;
+
+  @doc("URL location of schema, identified by schema ID.")
+  @header("Schema-Id-Location")
+  schemaIdLocation: string;
+
+  @doc("References schema group.")
+  @header("Schema-Group-Name")
+  schemaGroupName: string;
+
+  @doc("References schema name.")
+  @header("Schema-Name")
+  schemaName: string;
+
+  @doc("Version of the returned schema.")
+  @header("Schema-Version")
+  schemaVersion: int32;
+}
+
+@doc("Object received from the registry containing schema identifiers.")
+@resource("$schemaGroups/$schemas")
+model SchemaId {
+  @key("id")
+  @doc("Schema ID that uniquely identifies a schema in the registry namespace.")
+  @visibility("read")
+  id: string;
+}
+
+/**
+ * The list of strings with server paging support.
+ */
+@Azure.Core.pagedResult
+model StringsList {
+  /** The collection of pageable items. */
+  @Azure.Core.items
+  value: string[];
+
+  /** The link to the next page of items */
+  @Azure.Core.nextLink
+  nextLink?: string;
+}
+
+/**
+ * The list of int32s with server paging support.
+ */
+@Azure.Core.pagedResult
+model Int32sList {
+  /** The collection of pageable items. */
+  @Azure.Core.items
+  value: int32[];
+
+  /** The link to the next page of items */
+  @Azure.Core.nextLink
+  nextLink?: string;
+}
+
+@doc("Schema Group resource.")
+@resource("$schemaGroups")
+model SchemaGroup {
+  @key("groupName")
+  @doc("Name of schema group.")
+  @visibility("read")
+  groupName: string;
+}
+
+@doc("Schemas resource.")
+@parentResource(SchemaGroup)
+@resource("schemas")
+model SchemaName {
+  @key("name")
+  @doc("Name of schema.")
+  @visibility("read")
+  @maxLength(50)
+  @pattern("^[A-Za-z0-9][^\\\\/$:]*$")
+  name: string;
+}
+
+@doc("Schema versions resource.")
```

Comment:
try this:
model SchemaVersion {
  **@body**
  @key
  @visbility("read")
  id: int32;
}

Reply:
results in the following error:
`Param version has multiple types: [path, body]TypeSpec(@typespec/http/operation-param-duplicate-type`